### PR TITLE
auto dump pprof when cpu is high

### PR DESCRIPTION
**Pull request by zhshy11**
auto dump pprof when cpu is high

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/config/pprof.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			holmes.WithDumpPath(c.dumpPath),
+			holmes.WithCPUDump(c.cpuTriggerPercentMin, c.cpuTriggerPercentDiff, c.cpuTriggerPercentAbs),
+			holmes.WithMemDump(c.memTriggerPercentMin, c.memTriggerPercentDiff, c.memTriggerPercentAbs),
+			holmes.WithBinaryDump(),
+		)
+		if err != nil {
+			tmos.Exit(err.Error())
+		}
+		h.EnableCPUDump()
+		h.EnableMemDump()
```
Comment (ylsGit) [inline app/config/pprof.go]: h.EnableCPUDump().EnableMemDump()

File: app/config/pprof.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+func PprofDown() {
+	c := LoadPprofFromConfig()
+	fmt.Println(fmt.Sprintf("LoadPprofFromConfig = %v", c))
+	if c.autoDump {
+		h, err := holmes.New(
+			holmes.WithCollectInterval(c.collectInterval),
+			holmes.WithCoolDown(c.coolDown),
+			holmes.WithDumpPath(c.dumpPath),
+			holmes.WithCPUDump(c.cpuTriggerPercentMin, c.cpuTriggerPercentDiff, c.cpuTriggerPercentAbs),
+			holmes.WithMemDump(c.memTriggerPercentMin, c.memTriggerPercentDiff, c.memTriggerPercentAbs),
```
Comment (ylsGit) [inline app/config/pprof.go]: add   holmes.WithGoroutineDump()

File: app/config/pprof.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	FlagPprofCpuTriggerPercentAbs  = "pprof-cpu-trigger-percent-abs"
+	FlagPprofMemTriggerPercentMin  = "pprof-mem-trigger-percent-min"
+	FlagPprofMemTriggerPercentDiff = "pprof-mem-trigger-percent-diff"
+	FlagPprofMemTriggerPercentAbs  = "pprof-mem-trigger-percent-abs"
+)
+
+// PprofDown auto dump pprof
+func PprofDown() {
+	c := LoadPprofFromConfig()
+	fmt.Println(fmt.Sprintf("LoadPprofFromConfig = %v", c))
```
Comment (ylsGit) [inline app/config/pprof.go]: use  log.

File: app/config/pprof.go (hunk 1 above)
Comment (zhshy11) [reply app/config/pprof.go]: done

File: app/config/pprof.go (hunk 2 above)
Comment (zhshy11) [reply app/config/pprof.go]: done

File: app/config/pprof.go (hunk 3 above)
Comment (zhshy11) [reply app/config/pprof.go]: done

**Review by ylsGit — DISMISSED** [review]

**Review by ylsGit — DISMISSED** [review]

File: app/config/pprof.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	FlagPprofCpuTriggerPercentMin  = "pprof-cpu-trigger-percent-min"
+	FlagPprofCpuTriggerPercentDiff = "pprof-cpu-trigger-percent-diff"
+	FlagPprofCpuTriggerPercentAbs  = "pprof-cpu-trigger-percent-abs"
+	FlagPprofMemTriggerPercentMin  = "pprof-mem-trigger-percent-min"
+	FlagPprofMemTriggerPercentDiff = "pprof-mem-trigger-percent-diff"
+	FlagPprofMemTriggerPercentAbs  = "pprof-mem-trigger-percent-abs"
+)
+
+// PprofDown auto dump pprof
+func PprofDown() {
```
Comment (ilovers) [inline app/config/pprof.go]:
```suggestion
func PprofDownload() {
```

File: app/config/config.go [hunk 5]
```diff
@@ -64,6 +64,9 @@ func RegisterDynamicConfig() {
 	// set the dynamic config
 	oecConfig := GetOecConfig()
 	tmconfig.SetDynamicConfig(oecConfig)
+
+	//down pprof
+	PprofDown()
```
Comment (ilovers) [inline app/config/config.go]: Use a separate flag to determine whether to enable pprof automatic download.

File: app/config/pprof.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+	c := LoadPprofFromConfig()
+	if c.autoDump {
```
Comment (ilovers) [inline app/config/pprof.go]:
```suggestion
	if !c.autoDump {
	    return
	 }
```

**Review by ilovers — APPROVED** [review]

**Pull request merged**